posthog/performance/schema.py

```diff
@@ -0,0 +1,185 @@
+"""https://developer.mozilla.org/en-US/docs/Web/API/PerformanceEntry"""
+from posthog import settings
+from posthog.clickhouse.kafka_engine import KAFKA_COLUMNS_WITH_PARTITION, STORAGE_POLICY, kafka_engine
+from posthog.clickhouse.table_engines import Distributed, ReplacingMergeTree, ReplicationScheme
+from posthog.kafka_client.topics import KAFKA_PERFORMANCE_EVENTS
+
+# TODO
+# explode server timing from Resource events into their own columns
+# /* time origin is in milliseconds e.g. 1670900799301.7 */
+
+"""
+# expected queries
+
+## get all performance events for a given team's session
+
+allows us to show performance events alongside other logs while viewing a session recording
+
+shard by session id so that when querying for all in session or pageview all data for the results are on the same shard
+
+SELECT * FROM performance_events
+WHERE team_id = 1
+AND session_id = 'my-session-uuid'
+ORDER BY timestamp DESC
+
+## get all performance events for a given team's pageview
+
+allows us to show performance events in a waterfall chart for a given pageview
+
+SELECT * FROM performance_events
+WHERE team_id = 1
+AND pageview_id = 'my-page-view-uuid' -- sent by SDK
+ORDER BY timestamp DESC
+
+## all other queries are expected to be based on aggregating materialized views built from this fact table
+"""
+
+BASE_PERFORMANCE_EVENT_COLUMNS = """
```

Comment:
Nit: I think the extra constants make this _harder_ to follow rather than easier. 

But snapshots of `schema.py` (currently unupdated) will do the trick here.

Reply:
I hadn't seen the snapshots... that made developing/debugging the SQL creation way easier 🙌